operator/roles/kiali-deploy/tasks/main.yml

```diff
@@ -12,6 +12,7 @@
   set_fact:
     is_openshift: "{{ True if 'route.openshift.io' in api_groups else False }}"
     is_k8s: "{{ False if 'route.openshift.io' in api_groups else True }}"
+    is_maistra: "{{ True if 'maistra.io' in api_groups else False }}"
```

Comment:
you are going to want to set this separately from is_openshift/is_k8s... look at the "when" clause - this only is executed when is_openshift and is_k8s are both false. This is almost always going to be the case, but not 100% of the time, mainly for testing/development when an operator developer wants to forcibly set one of these to true in the playbook (which I do rarely, but it is done).

So remove this line 15 that was added, and simply add a new task just below... so you want something like this:

```
- name: Determine the Istio implementation
  set_fact:
￼   is_maistra: "{{ True if 'maistra.io' in api_groups else False }}"
```